phoenix-core/src/main/java/org/apache/phoenix/index/IndexMaintainer.java

```diff
@@ -883,6 +903,14 @@ public void readFields(DataInput input) throws IOException {
         dataEmptyKeyValueCF = Bytes.readByteArray(input);
         emptyKeyValueCFPtr = new ImmutableBytesPtr(Bytes.readByteArray(input));
         
+        int numIndexedExpressions = WritableUtils.readVInt(input);
+        indexedExpressions = Sets.newLinkedHashSetWithExpectedSize(numIndexedExpressions);        
+        for (int i = 0; i < numIndexedExpressions; i++) {
```

Comment:
indexedExpressions contain both regular columns and expressions. For regular columns the expression will be a ColumnExpression.